### PR TITLE
Fix model group issues

### PR DESCRIPTION
**Pull request by elihart**
Should fix https://github.com/airbnb/epoxy/issues/672 and https://github.com/airbnb/epoxy/issues/675

Tests were updated to catch these issues

### CODE REVIEW

File: epoxy-adapter/src/main/java/com/airbnb/epoxy/ModelGroupHolder.kt [hunk 1]
```diff
@@ -58,40 +58,31 @@ class ModelGroupHolder : EpoxyHolder() {
     }
 
     private fun createViewStubData(viewGroup: ViewGroup): List<ViewStubData> {
-        val stubs = ArrayList<ViewStubData>(4)
+        return ArrayList<ViewStubData>(4).apply {
```
Comment (eboudrant) [inline epoxy-adapter/src/main/java/com/airbnb/epoxy/ModelGroupHolder.kt]: Just curious, why don't you use Kotlin list : `mutableListOf<ViewStubData>()` ?

File: epoxy-adapter/src/main/java/com/airbnb/epoxy/ModelGroupHolder.kt [hunk 2]
```diff
@@ -217,7 +210,12 @@ private class ViewStubData(
         }
     }
 
-    fun removeView() {
+    fun resetStub() {
```
Comment (eboudrant) [inline epoxy-adapter/src/main/java/com/airbnb/epoxy/ModelGroupHolder.kt]: `private` ?

File: epoxy-adapter/src/main/java/com/airbnb/epoxy/ModelGroupHolder.kt (hunk 1 above)
Comment (elihart) [reply epoxy-adapter/src/main/java/com/airbnb/epoxy/ModelGroupHolder.kt]: I believe it was so I could give an initial capacity value

File: epoxy-adapter/src/main/java/com/airbnb/epoxy/ModelGroupHolder.kt (hunk 2 above)
Comment (elihart) [reply epoxy-adapter/src/main/java/com/airbnb/epoxy/ModelGroupHolder.kt]: it's actually used in the holder class in the same file, but ViewStubData is private

Comment (eboudrant) [reply epoxy-adapter/src/main/java/com/airbnb/epoxy/ModelGroupHolder.kt]: ah, could reduce to `internal` then ;)

File: epoxy-adapter/src/main/java/com/airbnb/epoxy/ModelGroupHolder.kt (hunk 1 above)
Comment (eboudrant) [reply epoxy-adapter/src/main/java/com/airbnb/epoxy/ModelGroupHolder.kt]: Ok, why 4 ?

Comment (elihart) [reply epoxy-adapter/src/main/java/com/airbnb/epoxy/ModelGroupHolder.kt]: it's a somewhat arbitrary guess - groups are generally small, but non zero size

**Pull request merged**